apps/judicial-system/backend/src/app/modules/case-table/whereOptions/prosecutor.ts

```diff
@@ -0,0 +1,101 @@
+import { fn, Op } from 'sequelize'
+
+import {
+  CaseAppealState,
+  CaseState,
+  completedIndictmentCaseStates,
+  completedRequestCaseStates,
+  indictmentCases,
+  investigationCases,
+  restrictionCases,
+  type User as TUser,
+} from '@island.is/judicial-system/types'
+
+const prosecutorSharedWhereOptions = (user: TUser) => ({
+  [Op.or]: [
+    { creating_prosecutor_id: user.id },
+    { prosecutor_id: user.id },
```

Comment:
The creating prosecutor id and prosecutor id should be grouped with the heightened securoty level check in an or clause. See `cases.filter.ts` for reference. These checks only apply to request cases. For indictment cases it is enough to verify the prosecutors office id.